machine_tool_rental/data/ir_model_fields.xml

```diff
@@ -0,0 +1,307 @@
+    <field name="name">Available</field>
+    <field name="field_id" ref="stock_model_stock_storage_category_x_serial_status_field" />
+    <field name="value">available</field>
+  </record>
+  <record id="stock_model_stock_storage_category_x_serial_status_selection_out" model="ir.model.fields.selection">
+    <field name="name">Out</field>
+    <field name="field_id" ref="stock_model_stock_storage_category_x_serial_status_field" />
+    <field name="value">out</field>
+  </record>
+  <record id="stock_model_stock_storage_category_x_serial_status_selection_reception" model="ir.model.fields.selection">
+    <field name="name">Reception</field>
+    <field name="field_id" ref="stock_model_stock_storage_category_x_serial_status_field" />
+    <field name="value">reception</field>
+  </record>
+  <record id="stock_model_stock_storage_category_x_serial_status_selection_maintenance" model="ir.model.fields.selection">
+    <field name="name">Maintenance</field>
+    <field name="field_id" ref="stock_model_stock_storage_category_x_serial_status_field" />
+    <field name="value">maintenance</field>
+  </record>
+  <record id="stock_model_stock_lot_x_status_field" model="ir.model.fields">
+    <field name="compute"><![CDATA[for record in self:
+    if record.location_id.storage_category_id and record.location_id.storage_category_id.x_serial_status:
+        record['x_status'] = record.location_id.storage_category_id.x_serial_status
+    else: record['x_status'] = "out"
+    ]]></field>
+    <field name="ttype">char</field>
+    <field name="depends">location_id</field>
+    <field name="field_description">Status</field>
+    <field name="model_id" ref="stock.model_stock_lot"/>
+    <field name="name">x_status</field>
+    <field name="readonly" eval="True"/>
+    <field name="selectable" eval="False"/>
+    <field name="store" eval="False"/>
+  </record>
+  <record id="to_invoice_total_uninvoiced_balance" model="ir.model.fields">
+    <field name="compute"><![CDATA[
+for record in self:
+    record['x_amount_to_invoice'] = 0
+    for so in record.sale_order_ids.filtered(lambda so: so.invoice_status == 'to invoice'):
+        if so.is_rental_order:
+            rental_amount = sum(so.order_line.filtered('is_product_rentable').mapped('amount_to_invoice'))
+            record['x_amount_to_invoice'] += so.amount_to_invoice - rental_amount
+            last_date = so.x_last_invoice or so.rental_start_date.date()
+            next_date = min(so.rental_return_date, datetime.datetime.today()).date()
+            record['x_amount_to_invoice'] += rental_amount * (next_date - last_date).days / so.duration_days
+        else: record['x_amount_to_invoice'] += sum(so.order_line.mapped('amount_to_invoice'))
+]]></field>
+    <field name="ttype">monetary</field>
+    <field name="field_description">Total Uninvoiced</field>
+    <field name="depends">sale_order_ids</field>
```

Comment:
Not sure you need a dependency here. Anytime the field is displayed, it has to be recomputed in my understandig